### DISCUSSION TITLE
Automated checklists for manual release steps

### DISCUSSION THREAD

**Discussion by gnodar01** [CellProfiler/CellProfiler#4943] [Ideas]
Our release process currently relies on a series of manual steps. I don't think we're likely to change that any time soon, but we could at least automatically generate checklists depending on what we're doing.

The *CellProfiler (and friends) release steps* google doc is okay-ish, but feels disconnected form everything else we use for managing CellProfiler stuff.

I set up a proof of concept in the private repo `CellProfiler/releases`. Nothing in there is sensitive information, but feels unnecessary to give broad public access to a very internal process.

The flow is this:

We submit a new issue

<img width="970" alt="Screenshot 2024-09-18 at 4 02 46 PM" src="https://github.com/user-attachments/assets/99582f06-4536-4124-b210-7a2eaa460de4">

---

The issue uses the new Github issue forms.

<img width="976" alt="Screenshot 2024-09-18 at 4 03 45 PM" src="https://github.com/user-attachments/assets/406d1921-7fb1-4100-88b7-885026171715">

---

You simply fill out the release type

<img width="705" alt="Screenshot 2024-09-18 at 4 04 09 PM" src="https://github.com/user-attachments/assets/2785a444-6cf8-45fb-9710-7344d86233e5">

---

Template Here

<img width="681" alt="Screenshot 2024-09-18 at 4 04 34 PM" src="https://github.com/user-attachments/assets/3f0d89e3-94a8-4477-88e9-1fceb5298b44">

---

Submitting the issue kicks off an action

<img width="442" alt="Screenshot 2024-09-18 at 4 04 51 PM" src="https://github.com/user-attachments/assets/ca29117c-7ee0-4b10-b91e-557a7168ba5f">

---

It parses the information

<img width="425" alt="Screenshot 2024-09-18 at 4 05 04 PM" src="https://github.com/user-attachments/assets/242fb967-d4b4-4da4-b17e-70e3c098d663">

---

And generates a description of what to do, and a checklist

<img width="377" alt="Screenshot 2024-09-18 at 4 05 24 PM" src="https://github.com/user-attachments/assets/eb780a04-fcd7-43dc-88bf-369efba5847e">

---

The checklist is recognized by GH as a task list which appears in the UI

<img width="603" alt="Screenshot 2024-09-18 at 4 05 58 PM" src="https://github.com/user-attachments/assets/1ccdb164-40c8-4f39-8231-c620ce05c603">

---

An example, after having run the action: https://github.com/CellProfiler/release/issues/11

**Comment by gnodar01** [comment]
@bethac07  Is this something you find useful?

I definitely plan on completing this, at the very least for myself to use. But before I continue, do you have any opinions to make it more broadly useful? Otherwise I'll end up tailoring it for my need specifically.

**Reply by bethac07** [reply]
Yes, totally useful, thanks much!
